examples/twittermap/web/app/controllers/TwitterMapApplication.scala

```diff
@@ -290,6 +302,49 @@ object TwitterMapApplication {
     }
   }
 
+  /** Find cities' population whose cityID are in cityIDs.
+    *
+    * @param cityIds List of cities in current boundary in String
+    * @param citiesPopulation List of all cities' population data
+    * @return List of cities population which centroids is in current boundary
+    */
+  def findCityPop(cityIds: String, citiesPopulation: List[JsValue]): JsArray = {
+    val cityIdsList : List[Int] = cityIds.split(",").map(_.toInt).toList
+    val incCityIdsList = cityIdsList.sortWith(_<_)
+    sortedListBinarySearch(citiesPopulation, 0, 29833, incCityIdsList, 0, Json.arr())
+  }
+
+  /**
+    * Find the population data data for each geoId in from the sorted geoIds list.
+    *
+    * @param sortedCityIds Sorted list of cities in current boundary
+    * @param citiesPopulation List of all cities' population data
+    * @return List of cities in sortedCityIds' population data
+    */
+  def sortedListBinarySearch(citiesPopulation: List[JsValue], startIndex: Int, endIndex: Int,
+                             sortedCityIds: List[Int], sortedCityIndex: Int, resultArr: JsArray): JsArray = {
+    if (startIndex == endIndex) {
+        resultArr
+    } else {
+      val thisIndex = (startIndex + endIndex) / 2
+      val thisCityId = (citiesPopulation(thisIndex) \ "cityID").as[Int]
+      if (thisCityId > sortedCityIds(sortedCityIndex)) {
+        sortedListBinarySearch(citiesPopulation, startIndex, thisIndex, sortedCityIds, sortedCityIndex, resultArr)
+      } else if (thisCityId < sortedCityIds(sortedCityIndex)) {
+        sortedListBinarySearch(citiesPopulation, thisIndex + 1, endIndex, sortedCityIds, sortedCityIndex, resultArr)
+      } else { // Current cityId sortedCityIds(sortedCityIndex) is find in citiesPopulation
+        val currPop : JsValue = JsObject(Seq("cityID" -> JsNumber(thisCityId),
+                                             "population" -> JsNumber((citiesPopulation(thisIndex) \ "population").as[Int])
+                                            ))
+        if (sortedCityIndex < sortedCityIds.length - 1) {
+          sortedListBinarySearch(citiesPopulation, thisIndex + 1, 29833, sortedCityIds, sortedCityIndex + 1, resultArr :+ currPop)
+        } else {
```

Comment:
```suggestion
        } else {
```
This is wrong, only the population data of the **last** city in the `sortedCityIds` can be appended to the `resultArr`.